### PR TITLE
Update useListData to handle select all w/ unloaded lists

### PR DESCRIPTION
**Pull request by jluyau**
If list data has unloaded items, removing the last visible selected item will keep the selection at the `all` value. If all items have been loaded, removing the last visible selected item will change selection to an empty set.

If list data has unloaded items, a remove `all` selected items will remain at the `all` value. If all items have been loaded, a remove `all` will change selection to an empty set.

### CODE REVIEW

File: packages/@react-stately/data/src/useListData.ts [hunk 1]
```diff
@@ -232,10 +236,11 @@ export function createListActions<T>(opts: ListOptions<T>, dispatch: (updater: (
     removeSelectedItems() {
       dispatch(state => {
         if (state.selectedKeys === 'all') {
+          // return 'all' value if there still exists unloaded items (i.e. cursor is not null)
           return {
             ...state,
             items: [],
-            selectedKeys: new Set()
+            selectedKeys: cursor != null ? 'all' : new Set()
```
Comment (devongovett) [inline packages/@react-stately/data/src/useListData.ts:243]: Wouldn't the previous behavior make sense here though? If `all` was previously selected, and those items were removed, then that means that all items, even those not yet loaded have been deleted.

File: packages/@react-stately/data/src/useListData.ts [hunk 2]
```diff
@@ -218,7 +222,7 @@ export function createListActions<T>(opts: ListOptions<T>, dispatch: (updater: (
             selection.delete(key);
           }
         }
-        if (selection === 'all' && items.length === 0) {
+        if (cursor == null && items.length === 0) {
```
Comment (devongovett) [inline packages/@react-stately/data/src/useListData.ts:225]: Should we add a test for this case?

File: packages/@react-stately/data/src/useListData.ts [hunk 3]
```diff
@@ -232,6 +236,7 @@ export function createListActions<T>(opts: ListOptions<T>, dispatch: (updater: (
     removeSelectedItems() {
       dispatch(state => {
         if (state.selectedKeys === 'all') {
+          // return 'all' value if there still exists unloaded items (i.e. cursor is not null)
```
Comment (LFDanLu) [inline packages/@react-stately/data/src/useListData.ts:239]: nit: remove this comment

**Review by LFDanLu — CHANGES_REQUESTED** [review]
Just a small nit, but otherwise LGTM

**Review by devongovett — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**